cf_il/main.py

```diff
@@ -0,0 +1,128 @@
+from argparse import ArgumentParser, Namespace
+
+import torch
+import torch.nn as nn
+from cf_il.model import CFIL
+from cf_il.train import train
+
+from datasets import NAMES as DATASET_NAMES, get_dataset
+from datasets.utils.continual_dataset import ContinualDataset
+
+parser = ArgumentParser(description='cf-lr', allow_abbrev=False)
+
+# Logging
+parser.add_argument(
+    '--tensorboard',
+    action='store_true',
+    help='Enable tensorboard logging',
+)
+
+# Dataset
+parser.add_argument(
+    '--dataset',
+    type=str,
+    required=True,
+    choices=DATASET_NAMES,
+    help='Which dataset to perform experiments on.',
+)
+parser.add_argument(
+    '--validation',
+    action='store_true',
+    help='Test on the validation set',
+)
+
+# Experiments setup
+parser.add_argument(
+    '--lr',
+    type=float,
+    required=True,
+    help='Learning rate.',
+)
+parser.add_argument(
+    '--momentum',
+    type=float,
+    required=True,
+    help='Momentum.',
+)
+parser.add_argument(
+    '--batch_size',
+    type=int,
+    required=True,
+    help='Batch size.',
+)
+parser.add_argument(
+    '--n_epochs',
+    type=int,
+    required=True,
+    help='The number of epochs for each task.',
+)
+
+# Memory buffer
+parser.add_argument(
+    '--buffer_size',
+    type=int,
+    required=True,
+    help='The size of the memory buffer.',
+)
+parser.add_argument(
+    '--minibatch_size',
+    type=int,
+    required=True,
+    help='The batch size of the memory buffer.',
+)
+
+# CF-IL
+parser.add_argument(
+    '--alpha',
+    type=float,
+    required=True,
+    help='Penalty weight.',
```

Comment:
może lepiej w opisie dać "degree of distilation" jak jest w artykule?

Reply:
Faktycznie, bardziej to pasuje